slsim/Sources/SourceVariability/variability.py

```diff
@@ -13,32 +16,36 @@ def __init__(self, variability_model, **kwargs_variability_model):
             sinusoidal_variability kwargs are amplitude ('amp') and frequency ('freq').
         :type kwargs_variability_model: dict
         """
+        self.kwargs_model = kwargs_variability_model
         self.variability_model = variability_model
-        if self.variability_model not in ["sinusoidal"]:
-            raise ValueError(
-                "Given model is not supported. Currently supported model is sinusoidal."
-            )
         if self.variability_model == "sinusoidal":
-            self._model = sinusoidal_variability
+            sinusoidal_class = SinusoidalVariability(**self.kwargs_model)
+            self._model = sinusoidal_class.magnitude
+        elif self.variability_model == "light_curve":
```

Comment:
just to clarify, this class only supports a single band variability, correct?

Reply:
Yes, you are right.